columnflow/plotting/plot_util.py

```diff
@@ -1016,3 +1019,58 @@ def remove_label_placeholders(
         sel = "[A-Z0-9]+"
 
     return re.sub(f"__{sel}__", "", label)
+
+
+def calculate_error(
+    hist: hist.Hist,
+    error_type: str,
+) -> dict:
+    """
+    Calculate the error to be plotted for the given histogram *hist*.
+    Supported error types are:
+        - 'variance': the plotted error is the square root of the variance for each bin
+        - 'poisson_unweighted': the plotted error is the poisson error for each bin
+        - 'poisson_weighted': the plotted error is the poisson error for each bin, weighted by the variance
+    """
+    logger = law.logger.get_logger(__name__)
+
+    # determine the error type
+    if error_type == "variance":
+        yerr = hist.view().variance ** 0.5
+    elif error_type.startswith("poisson"):
+        # compute asymmetric poisson confidence interval
+        from hist.intervals import poisson_interval
+
+        if error_type == "poisson_unweighted":
+            variances = None
+        elif error_type == "poisson_weighted":
+            variances = hist.view().variance
+        else:
+            raise ValueError(f"unknown error type '{error_type}'")
```

Comment:
With the above change, this can be simplified to

```suggestion
        variances = hist.view().variance if error_type == "poisson_weighted" else None
```